src/cart/src/cart/crt_hg.c

```diff
@@ -1149,7 +1149,8 @@ crt_hg_req_send(struct crt_rpc_priv *rpc_priv)
 			  rpc_priv->crp_tgt_uri);
 	}
 
-	if (hg_ret == HG_NA_ERROR) {
+	if (hg_ret == HG_NA_ERROR ||
+		hg_ret == HG_PROTOCOL_ERROR) {
```

Comment:
A comment explaining when those errors can happen and how we process them would be welcomed.